File: 00-solution/civo/python/__main__.py
import pulumi
import pulumi_civo as civo

firewall = civo.Firewall("firewall",
    create_default_rules=True,
    region="LON1")
cluster = civo.KubernetesCluster("cluster",
    firewall_id=firewall.id,
    region="LON1",
    cni="cilium",
    applications="-traefik2-nodeport,civo-cluster-autoscaler,traefik2-loadbalancer",
    kubernetes_version="1.28.7-k3s1",
    write_kubeconfig=True,
    pools={
        "size": "g4s.kube.medium",
        "node_count": 1,
    })
pulumi.export("name", cluster.name)
pulumi.export("kubeconfig", cluster.kubeconfig)

File: 00-solution/civo/go/main.go
package main

import (
	"github.com/pulumi/pulumi-civo/sdk/v2/go/civo"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		firewall, err := civo.NewFirewall(ctx, "firewall", &civo.FirewallArgs{
			CreateDefaultRules: pulumi.Bool(true),
			Region:             pulumi.String("LON1"),
		})
		if err != nil {
			return err
		}
		cluster, err := civo.NewKubernetesCluster(ctx, "cluster", &civo.KubernetesClusterArgs{
			FirewallId:        firewall.ID(),
			Region:            pulumi.String("LON1"),
			Cni:               pulumi.String("cilium"),
			Applications:      pulumi.String("-traefik2-nodeport,civo-cluster-autoscaler,traefik2-loadbalancer"),
			KubernetesVersion: pulumi.String("1.28.7-k3s1"),
			WriteKubeconfig:   pulumi.Bool(true),
			Pools: &civo.KubernetesClusterPoolsArgs{
				Size:      pulumi.String("g4s.kube.medium"),
				NodeCount: pulumi.Int(1),
			},
		})
		if err != nil {
			return err
		}
		ctx.Export("name", cluster.Name)
		ctx.Export("kubeconfig", cluster.Kubeconfig)
		return nil
	})
}
